### PR TITLE
Resolve test failure from bundler 2.2.24+

### PR DESCRIPTION
**Pull request by iterateNZ**
Bundler v2.2.24 changed the behaviour of `Bundler::Definition#specs_for` when the requested groups are empty from returning nothing to everything.

### CODE REVIEW

**Comment by enebo** [comment]
Just poking in to note that jruby-9.2 is pretty ancient so it failing with the new .lock file is not so surprising.  This work should probably update to 9.4 (9.3 is still supported but not for too much longer).    Not sure if that will start to pull the sweater apart or be relatively error-free.

**Comment by iterateNZ** [comment]
Trying a slightly different tack and including two bundler versions in the build matrix (sneaking in jruby 9.3, 9.4 and java 11 as well). I haven't been able to get clean consistent builds of these environments locally so relying on the CI setup here to verify the changes.

The bundler version running the rspec should be what is then used for the `bundle install` within the examples upgrading the lock file as it goes 🤞

File: spec/warbler/web_server_spec.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def self.reset_local_repository
 
 describe Warbler::WebServer::Artifact do
 
-  @@_env = ENV.dup
+  @@_env = ENV.to_h
```
Comment (iterateNZ) [inline spec/warbler/web_server_spec.rb:11]: Should resolve test failures on jruby v9.4+

**Comment by iterateNZ** [comment]
Looks like this is passing on jruby-head now as well, that's a nice bonus.

**Pull request merged**